wdl-analysis/src/diagnostics.rs

```diff
@@ -792,3 +792,11 @@ pub fn invalid_placeholder_option<N: TreeNode>(
 
     Diagnostic::error(message).with_label(format!("this is type `{ty}`"), span)
 }
+
+/// Creates an invalid regex pattern diagnostic.
+pub fn invalid_regex_pattern(function: &str, pattern: &str, error: &str, span: Span) -> Diagnostic {
```

Comment:
We can probably have `error` be `&regex::Error` as we don't need to create a string from it.